scripts/build

```diff
@@ -265,6 +264,36 @@ if [ ! -f $STAMP ]; then
         PKG_MESON_SCRIPT="$PKG_BUILD/meson.build"
       fi
 
+      # auto detect toolchain
+      _auto_toolchain=""
+      if [ -z "$PKG_TOOLCHAIN" -o "$PKG_TOOLCHAIN" = "auto" ]; then
+        if [ -f "$PKG_MESON_SCRIPT" ]; then
+          PKG_TOOLCHAIN="meson"
+        elif [ -f "$PKG_CMAKE_SCRIPT" -a ! "$PKG_USE_CMAKE" = "no" ]; then
+          PKG_TOOLCHAIN="cmake"
+        elif [ -f "$PKG_CONFIGURE_SCRIPT" ]; then
+          PKG_TOOLCHAIN="configure"
+        elif [ -f $PKG_BUILD/Makefile ]; then
+          PKG_TOOLCHAIN="make"
+        else
+          echo "Not possible to detect toolchain automactcly. Add PKG_TOOLCHAIN= in to package.mk"
+          exit 1
+        fi
+        _auto_toolchain=" (auto-detect)"
+      fi
+      if [ $PKG_TOOLCHAIN != "meson" -a $PKG_TOOLCHAIN != "cmake" -a 
+           $PKG_TOOLCHAIN != "cmake-make" -a $PKG_TOOLCCHAIN != "configure" -a 
+           $PKG_TOOLCHAIN != "make" -a $PKG_TOOLCHAIN != "autotools" -a 
+           $PKG_TOOLCHAIN != "manual" ]; then
+        printf "$(print_color bold-red "ERROR: ")unknown toolchain $PKG_TOOLCHAIN"
+        exit 1
+      fi
+      printf "%${BUILD_INDENT}c $(print_color CLR_TOOLCHAIN "TOOLCHAIN    ")$PKG_TOOLCHAIN${_auto_toolchain}\n" ' '>&$SILENT_OUT
```

Comment:
Nit - stylistically this is inconsistent with existing usage, eg.:
```
printf "%${BUILD_INDENT}c $(print_color CLR_UNPACK "UNPACK")   $1\n" ' '>&$SILENT_OUT
```
The spaces follow the call to `print_color`, so your line should be:
```
-      printf "%${BUILD_INDENT}c $(print_color CLR_TOOLCHAIN "TOOLCHAIN    ")$PKG_TOOLCHAIN${_auto_toolchain}\n" ' '>&$SILENT_OUT 
+      printf "%${BUILD_INDENT}c $(print_color CLR_TOOLCHAIN "TOOLCHAIN")    $PKG_TOOLCHAIN${_auto_toolchain}\n" ' '>&$SILENT_OUT
```